File: Data Structures & Algorithms/two-integer-sum-ii/submission-3.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:

        l = 0
        h = len(numbers) - 1

        res = []

        while l < h:
            _sum = numbers[l] +  numbers[h]

            if _sum > target:
                h -= 1
            elif _sum < target:
                l += 1
            else:
                res.append(l+1)
                res.append(h+1)
                return res
        
        return res

File: Data Structures & Algorithms/two-integer-sum/submission-3.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> numberMap = new HashMap<>();

        for(int i = 0; i < nums.length; i++) {
            int currentNumber = nums[i];
            int complement = target - currentNumber;

            if(numberMap.containsKey(complement)) {
                return new int[] {numberMap.get(complement), i};
            }
            numberMap.put(currentNumber, i);
        }
        return null;
    }
}
